app/policies/degrees/trainee_policy.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Degrees::TraineePolicy < TraineePolicy
```

Comment:
It's more of a policy on degree rather then trainee

Reply:
This is how it is currently setup on `main`. It uses the [`TraineePolicy`](https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/policies/trainee_policy.rb) from [base_controller.rb#L16](https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/controllers/trainees/base_controller.rb#L16)

All I'm doing here is overriding the permissions on `#new` and `#create` for `degrees` and name it `Degrees::TraineePolicy` for consistency with the `TraineePolicy` which is based on.

Reply:
I think I agree with both points being made here! It looks like we are making a change to the policy for Degrees, but the way policies are currently done is centred on the Trainee.

I think this raises 2 broader questions about how policies are implemented, and what actions we want admins to be able to do.

For now, I think the approach here is ok, although I notice that a slightly different pattern is used for placements, and it would be good to keep things consistent:
* https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/controllers/trainees/placements_controller.rb#L80-L82
* https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/policies/trainee_policy.rb#L112-L116

Reply:
My thinking behind this is just based on [documentation](https://github.com/varvet/pundit?tab=readme-ov-file#policies), which gives it a better trod path

> The class has the same name as some kind of model class, only suffixed with the word "Policy".

Hence `DegreePolicy`

> The class implements some kind of query method, in this case update?. Usually, this will map to the name of a particular controller action.

Which in turns explains the overriding

> All I'm doing here is overriding the permissions on #new and #create for degrees

Reply:
@defong I agree that ideally each model should have each own Policy. The thing is the code has been structured in such a way that uses the `TraineePolicy` for all associated entities in https://github.com/DFE-Digital/register-trainee-teachers/blob/main/app/controllers/trainees/base_controller.rb#L7 including the `Degrees` and my thinking was to reuse what we got instead of rewriting the whole implementation. 

Do you prefer to skip the `before_action :authorize_trainee` in `Trainees::BaseController` for `Trainees::DegreesController` and use a `DegreePolicy` instead for this controller?

Reply:
Had a discussion IRL, this looks like early sign of tech debt without enough substance for now.

We will have another card to review and address current policy tech debt, as that will be more impactful work.